### PR TITLE
Add types for `crack-json`

### PR DESCRIPTION
**Pull request by jneisener**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/crack-json/index.d.ts
1. types/crack-json/crack-json-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d52d77ed0c71673c05031d8709cfa27711dbadaf
</p>

File: types/crack-json/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "rules": {
+        "@definitelytyped/no-declare-current-package": "off",
+        "@definitelytyped/no-single-declare-module": "off"
+    }
+}
```
Comment (johenning) [inline types/crack-json/.eslintrc.json:6]: I'd appreciate some guidance if these exceptions are ok. I can't get it to work without `declare module "crack-json"`, but maybe I'm just doing it wrong

File: types/crack-json/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+declare module "crack-json" {
+    function extractJson(
+        subject: string,
+        configuration?: {
+            filter?: (input: string) => boolean;
+            parser?: (input: string) => any;
+        },
+    ): any;
```
Comment (johenning) [inline types/crack-json/index.d.ts:8]: The library is essentially just a wrapper around `JSON.parse`, which is also typed as returning `any`, so I suppose `any` is okay to use here

Comment (peterblazejewicz) [reply types/crack-json/index.d.ts:8]: export named function:
```ts
export function extractJson(....): any;
```
the module exports just named function (and eventually types for configuration object)

Comment (peterblazejewicz) [reply types/crack-json/index.d.ts:8]: https://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/crack-json/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 750c9b9a0579bf235c8daf9e5b8cc1f487e72326
</p>

File: types/crack-json/index.d.ts (hunk 2 above)
Comment (johenning) [reply types/crack-json/index.d.ts]: Thanks. Turns out my local install wasn't working right, thats why I couldn't get it to work wihout the `declare module`

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@johenning thanks!

**Comment by johenning** [comment]
Ready to merge

**Pull request merged**